maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java

```diff
@@ -82,48 +92,174 @@
     }
 
     /**
-     * Represents a numeric item in the version item list.
+     * Represents a numeric item in the version item list that can be represented with an int.
      */
-    private static class IntegerItem
+    private static class IntItem
         implements Item
     {
-        private static final BigInteger BIG_INTEGER_ZERO = new BigInteger( "0" );
+        private final int value;
 
-        private final BigInteger value;
+        public static final IntItem ZERO = new IntItem();
+
+        private IntItem()
```

Comment:
This constructor feels superfluous.

Reply:
It construct a reusable Item instance representing a zero in a version string. If deleted, that instance must be constructed using an unnecessary string parse: new IntItem("0").

Reply:
That seems trivial performance wise and would be clearer.

Reply:
MNG-6572 is all about improving a performance problem and that constructor contributes to that objective. May we accord on adding a comment explaining its purpose?

Reply:
I would need to see careful benchmarks proving this is a hotspot first; and if the result was even measurable I'd be very surprised. This is a constructor that will be called once to set a single static field. This sort of microoptimization is almost never worth any reduction in clarity.